syntax = "proto3";

package Dictype;

service Dictype {
  rpc Transcribe(TranscribeRequest) returns (stream TranscribeResponse);
  rpc Stop(StopRequest) returns (StopResponse);
}

message TranscribeRequest {
  string profile_name = 1; // which predefined profile to use
}

message TranscribeResponse {
  string text = 1;
  uint32 begin_time = 2;
  bool sentence_end = 3;
}

message StopRequest {}

message StopResponse {
  bool stopped = 1;
}
